### PR TITLE
Update Grid and Pivot Visualizations

### PR DESCRIPTION
**Pull request by recca5p**

### CODE REVIEW

File: src/Reveal.Sdk.Dom.Tests/Reveal.Sdk.Dom.Tests.csproj [hunk 1]
```diff
@@ -67,11 +67,4 @@
     </PackageReference>
   </ItemGroup>
 
-  <ItemGroup>
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Reveal.Sdk.Dom.Tests.csproj:70]: Please recheck this @recca5p

File: src/Reveal.Sdk.Dom.Tests/Visualizations/GridVisualizationFixture.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Reveal.Sdk.Dom.Tests.Visualizations
+{
+
+    public class GridVisualizationTests
+    {
+        [Fact]
+        public void GridVisualization_DefaultConstructor_ShouldInitializeWithDefaultValues()
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/GridVisualizationFixture.cs:17]: Check again for the name of the UT

File: src/Reveal.Sdk.Dom.Tests/Visualizations/GridVisualizationFixture.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        {
+            // Act
+            var gridVisualization = new GridVisualization();
+
+            // Assert
+            Assert.NotNull(gridVisualization);
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/GridVisualizationFixture.cs:26]: Check if we need further Asserts, for ex, Title, dataSourceItem (b/c title, dataSourceItem updates are included in the codes of the constructor)

File: src/Reveal.Sdk.Dom.Tests/Visualizations/GridVisualizationFixture.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Reveal.Sdk.Dom.Tests.Visualizations
+{
+
+    public class GridVisualizationTests
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/GridVisualizationFixture.cs:17]: It seems that we are missing test for ` public GridVisualization(DataSourceItem dataSourceItem) : this(null, dataSourceItem) { }`

Comment (vulh-infragistics) [reply src/Reveal.Sdk.Dom.Tests/Visualizations/GridVisualizationFixture.cs:17]: Also check for the naming convensions. This should be `GridVisualizationFixture`

File: src/Reveal.Sdk.Dom.Tests/Visualizations/GridVisualizationFixture.cs [hunk 5]
```diff
@@ -0,0 +1,371 @@
...
+            Assert.Equal(title, gridVisualization.Title);
+            Assert.Equal(ChartType.Grid, gridVisualization.ChartType);
+            Assert.NotNull(gridVisualization.Columns);
+            Assert.Empty(gridVisualization.Columns);
+        }
+
+        [Theory]
+        [InlineData(true, 100)]
+        [InlineData(false, 50)]
+        public void GridVisualizationSettings_PageSettings_ShouldSetCorrectValues(bool pagingEnabled, int pageSize)
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/GridVisualizationFixture.cs:50]: This should be moved to another new fixture: `GridVisualizationSettingsFixture`

Comment (vulh-infragistics) [reply src/Reveal.Sdk.Dom.Tests/Visualizations/GridVisualizationFixture.cs:50]: Also, check if we need Unit test for those types of fields with simple getter and setter

File: src/Reveal.Sdk.Dom.Tests/Visualizations/GridVisualizationFixture.cs (hunk 3 above)
Comment (recca5p) [reply src/Reveal.Sdk.Dom.Tests/Visualizations/GridVisualizationFixture.cs:26]: For this, I think I will create other tests to verify which each constructor class to create a object, so for this constructor function, I only verify which it being created by that constructor

File: src/Reveal.Sdk.Dom.Tests/Visualizations/GridVisualizationFixture.cs (hunk 5 above)
Comment (recca5p) [reply src/Reveal.Sdk.Dom.Tests/Visualizations/GridVisualizationFixture.cs:50]: Oke I will create another test classes for those settings with another issue

File: src/Reveal.Sdk.Dom.Tests/Visualizations/PivotVisualizationFixture.cs [hunk 6]
```diff
@@ -0,0 +1,344 @@
...
+            var jObject = JObject.Parse(json);
+            var actualJArray = (JArray)jObject["Widgets"];
+            var expectedJArray = JArray.Parse(expectedJson);
+
+            // Assert
+            Assert.Equal(expectedJArray.Count, actualJArray.Count);
+
+            for (int i = 0; i < expectedJArray.Count; i++)
+            {
+                Assert.Equal(expectedJArray[i], actualJArray[i]);
```
Comment (hainv-ohio) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/PivotVisualizationFixture.cs:340]: This is not correct assertion. No way the UUID can be the same as in the expected! If you debug, you will see the json are not the same but the test still pass! Re-check pls

File: src/Reveal.Sdk.Dom.Tests/Visualizations/PivotVisualizationFixture.cs [hunk 7]
```diff
@@ -0,0 +1,344 @@
...
+            var document = new RdashDocument("My Dashboard");
+
+            var excelDataSourceItem = new RestDataSourceItem("Marketing Sheet")
+            {
+                Subtitle = "Excel Data Source Item",
+                Url = "http://dl.infragistics.com/reportplus/reveal/samples/Samples.xlsx",
+                IsAnonymous = true,
+                Fields = new List<IField>
+                {
+                    new DateField("Date"),
```
Comment (hainv-ohio) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/PivotVisualizationFixture.cs:292]: Don't need to have all the fields, make it shorter

File: src/Reveal.Sdk.Dom.Tests/Visualizations/GridVisualizationFixture.cs [hunk 8]
```diff
@@ -0,0 +1,358 @@
...
+            var jObject = JObject.Parse(json);
+            var actualJArray = (JArray)jObject["Widgets"];
+            var expectedJArray = JArray.Parse(expectedJson);
+
+            // Assert
+            Assert.Equal(expectedJArray.Count, actualJArray.Count);
+
+            for (int i = 0; i < expectedJArray.Count; i++)
+            {
+                Assert.Equal(expectedJArray[i], actualJArray[i]);
```
Comment (hainv-ohio) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/GridVisualizationFixture.cs:354]: Same here, recheck, this is not correct. Pls make sure the asserts are meaningful

**Review by hainv-ohio — CHANGES_REQUESTED** [review]

**Review by hainv-ohio — CHANGES_REQUESTED** [review]
Please include tests for fields, like Columns getter, VisualizationDataSpec in GridVisualization. Same for fields in Pivot Visualization.

**Review by hainv-ohio — CHANGES_REQUESTED** [review]

File: e2e/Sandbox/MainWindow.xaml.cs [hunk 9]
```diff
@@ -88,7 +88,6 @@ public MainWindow()
                 .RegisterAmazonS3()
                 .RegisterGoogleDrive()
                 .RegisterOracle();
```
[the remark this answers is not in the document]
Comment (recca5p) [reply e2e/Sandbox/MainWindow.xaml.cs:90]: It is being declared above

**Review by vulh-infragistics — CHANGES_REQUESTED** [review]

**Review by hainv-ohio — APPROVED** [review]

File: src/Reveal.Sdk.Dom.Tests/Visualizations/PivotVisualizationFixture.cs [hunk 10]
```diff
@@ -0,0 +1,406 @@
...
+        }
+    }
+    
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void Constructor_InitializesPivotVisualizationWithDataSource_WhenDataSourceItemIsProvided(bool hasTabularData)
+    {
+        // Arrange
+        var dataSourceItem = new DataSourceItem { HasTabularData = true };
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/PivotVisualizationFixture.cs:79]:
```suggestion
        var dataSourceItem = new DataSourceItem { HasTabularData = hasTabularData};
```

File: src/Reveal.Sdk.Dom.Tests/Visualizations/PivotVisualizationFixture.cs [hunk 11]
```diff
@@ -0,0 +1,406 @@
...
+
+        // Act
+        var pivotVisualization = new PivotVisualization(dataSourceItem);
+
+        // Assert
+        Assert.Equal(ChartType.Pivot, pivotVisualization.ChartType);
+        Assert.NotNull(pivotVisualization.Columns);
+        Assert.Empty(pivotVisualization.Columns);
+        Assert.Null(pivotVisualization.Title);
+        Assert.IsType<TabularDataDefinition>(pivotVisualization.DataDefinition);
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/PivotVisualizationFixture.cs:89]: Need to check `DataDefinition` type

File: src/Reveal.Sdk.Dom.Tests/Visualizations/PivotVisualizationFixture.cs [hunk 12]
```diff
@@ -0,0 +1,406 @@
...
+
+        // Act
+        var values = pivotVisualization.Values;
+
+        // Assert
+        Assert.Equal(expectedValues, values);
+    }
+
+    [Fact]
+    public void Columns_AreSerializedCorrectly_WhenSerialize()
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/PivotVisualizationFixture.cs:168]: Remove this unit test

**Review by vulh-infragistics — CHANGES_REQUESTED** [review]
